polyfill/lib/date.mjs

```diff
@@ -269,20 +270,21 @@ export class Date {
     if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
     const YearMonth = GetIntrinsic('%Temporal.YearMonth%');
     const calendar = GetSlot(this, CALENDAR);
-    const fields = ES.ToTemporalDateRecord(this);
+    const fields = ES.ToTemporalDateFields(this, calendar);
     return calendar.yearMonthFromFields(fields, {}, YearMonth);
   }
   toMonthDay() {
     if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
     const MonthDay = GetIntrinsic('%Temporal.MonthDay%');
     const calendar = GetSlot(this, CALENDAR);
-    const fields = ES.ToTemporalDateRecord(this);
+    const fields = ES.ToTemporalDateFields(this, calendar);
     return calendar.monthDayFromFields(fields, {}, MonthDay);
   }
   getFields() {
-    const fields = ES.ToTemporalDateRecord(this);
-    if (!fields) throw new TypeError('invalid receiver');
-    fields.calendar = GetSlot(this, CALENDAR);
+    if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
+    const calendar = GetSlot(this, CALENDAR);
+    const fields = ES.ToTemporalDateFields(this, calendar);
+    fields.calendar = calendar;
```

Comment:
What if fields() returns "calendar" as one of the fields?

Reply:
I'm not sure what should be done in that case; I guess one option is to throw if `calendar` is returned in the validation step I mentioned in https://github.com/tc39/proposal-temporal/pull/1014#discussion_r510403224. The other option is to :shrug: and say "well, don't do that then." I'm mildly in favour of the latter since I think it's in line with what we do elsewhere for custom calendars and time zones.

Reply:
→ #1047